inspectit-gepard-agent/src/main/java/rocks/inspectit/gepard/agent/internal/instrumentation/model/rules/scopes/InstrumentationScope.java

```diff
@@ -0,0 +1,101 @@
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import rocks.inspectit.gepard.agent.instrumentation.state.configuration.matcher.CustomElementMatchers;
+import rocks.inspectit.gepard.agent.instrumentation.state.configuration.matcher.MatcherChainBuilder;
+import rocks.inspectit.gepard.config.model.instrumentation.scopes.ScopeConfiguration;
+
+/** Internal model, which holds type and method matchers to instrument a scope. */
+public class InstrumentationScope {
+
+  /**
+   * The method matcher we use in inspectIT, that indicates that all methods should be matched.
+   * TODO: Currently, we exclude synthetic methods, since they cause a bug.
+   */
+  public static final ElementMatcher.Junction<MethodDescription> ALL_METHODS =
+      isMethod().and(not(isSynthetic()));
+
+  /** Matcher for types of this scope */
+  private final ElementMatcher.Junction<TypeDescription> typeMatcher;
+
+  /** Matcher for methods of this scope */
+  private final ElementMatcher.Junction<MethodDescription> methodMatcher;
+
+  private InstrumentationScope(
+      ElementMatcher.Junction<TypeDescription> typeMatcher,
+      ElementMatcher.Junction<MethodDescription> methodMatcher) {
+    this.typeMatcher = typeMatcher;
+    this.methodMatcher = methodMatcher;
+  }
+
+  /**
+   * @return the type matcher of this scope
+   */
+  public ElementMatcher.Junction<TypeDescription> getTypeMatcher() {
+    return typeMatcher;
+  }
+
+  /**
+   * @return the method matcher of this scope
+   */
+  public ElementMatcher.Junction<MethodDescription> getMethodMatcher() {
+    return methodMatcher;
+  }
+
+  /**
+   * Creates an {@link InstrumentationScope} out of a {@link ScopeConfiguration}
+   *
+   * @param scope the scope configuration
+   * @return the instrumentation scope
+   */
+  public static InstrumentationScope create(ScopeConfiguration scope) {
+    ElementMatcher.Junction<TypeDescription> typeMatcher = createTypeMatcher(scope);
+    ElementMatcher.Junction<MethodDescription> methodMatcher = createMethodMatcher(scope);
+
+    return new InstrumentationScope(typeMatcher, methodMatcher);
+  }
+
+  /**
+   * @param scope the scope configuration
+   * @return the type matcher for the scope (Currently, we just match the fqn of the class)
+   */
+  private static ElementMatcher.Junction<TypeDescription> createTypeMatcher(
+      ScopeConfiguration scope) {
+    return named(scope.getFqn());
+  }
+
+  /**
+   * @param scope the scope configuration
+   * @return the method matcher for the scope
+   */
+  private static ElementMatcher.Junction<MethodDescription> createMethodMatcher(
+      ScopeConfiguration scope) {
+    MatcherChainBuilder<MethodDescription> matcherChainBuilder = new MatcherChainBuilder<>();
+    List<String> methods = scope.getMethods();
+
+    if (methods.isEmpty()) return ALL_METHODS;
+
+    methods.forEach(methodName -> matcherChainBuilder.or(CustomElementMatchers.nameIs(methodName)));
```

Comment:
Currently, we cannot instrument abstract classes, thus also not abstract methods.

I just tested what happens, when we try to instrument native methods. Nothing...
Which makes sense, since they are not written in Java. Also the Ocelot agent is not able to instrument them.